File: app/src/FizzBuzzType3.php
<?php

namespace App;

class FizzBuzzType3 implements FizzBuzzType
{
    public function generate(int $number): string
    {
        if ($number % 3 === 0 && $number % 5 === 0) {
            return 'FizzBuzz';
        }
        return (string) $number;
    }
}

File: app/fizz_buzz/lib/types/fizz_buzz_type3.ex
# FizzBuzzタイプ3の実装（3と5で逆のルール）
defmodule FizzBuzzType3 do
  use FizzBuzzTypeBase

  defimpl FizzBuzzType do
    def generate(_type, number) do
      FizzBuzzTypeBase.basic_fizzbuzz(number, 5, 3, "Fizz", "Buzz")
    end
  end
end